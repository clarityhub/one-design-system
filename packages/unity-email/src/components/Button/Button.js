import React from 'react';
import {
	MjmlButton,
} from 'mjml-react';
import { node } from 'prop-types';
import colors from '@clarityhub/unity-core/dist/theme/colors';

import { variants } from '../../theme/fonts';

const { letterSpacing, ...validButtonOptions } = variants.button;

const buttonBase = Object.assign(
	{},
	validButtonOptions,
	{
		fontSize: '16px',
		borderRadius: '4px', // TODO merge with borders.borderRadius
		paddingTop: '8px',
		paddingBottom: '8px',
		paddingLeft: '20px',
		paddingRight: '20px',
	}
);

const determineColors = (type) => {
	switch (type) {
	case 'primary':
		return {
			color: colors.white,
			backgroundColor: colors.primary,
		};
	case 'danger':
		return {
			color: colors.white,
			backgroundColor: colors.danger,
		};
	case 'default':
	default:
		return {
			color: colors.primary,
			backgroundColor: 'rgba(255, 255, 255, 0)',
		};
	}
};

const Button = ({ children, center, type, ...rest }) => (
	<MjmlButton {...buttonBase} {...determineColors(type)} align={center ? 'center' : 'left'} textAlign={center ? 'left': 'center'} {...rest}>
		{children}
	</MjmlButton>
);

Button.propType = {
	children: node,
};

export default Button;
