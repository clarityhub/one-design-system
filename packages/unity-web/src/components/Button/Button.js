import React from 'react';
import cx from 'classnames';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool, func, oneOf, oneOfType, node, string } from 'prop-types';
import { variants } from '@clarityhub/unity-core/lib/typography';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';
import { lighten, darken, desaturate } from '@clarityhub/unity-core/lib/utilities/color';

import Loading from '../svgs/Loading';

const Button = styled.button(
	css`
        overflow: visible;
        display: inline-block;

        ${variants.button.string}

        box-shadow: ${colors.shadow.default};
        cursor: pointer;
        text-align: center;
        white-space: nowrap;
        vertical-align: middle;
        user-select: none;
        border: 1px solid transparent;
        padding: .5rem 1.25rem;
        font-size: 1rem;
        line-height: 1.5;
        border-radius: ${borders.borderRadius.rem}rem;
        position: relative;
        text-decoration: none;
        text-transform: uppercase;
        transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;

        &:focus,
        &:hover {
            text-decoration: none;
        }

        &:focus,
        &.focused {
            outline: 0;
            box-shadow: 0 0 0 .2rem ${colors.focused.default};
        }

        &:disabled,
        &.disabled {
            cursor: not-allowed;
            pointer-events: none;
        }

        &:not(:disabled):not(.disabled) {
            cursor: pointer;
        }

        svg {
            height: 1rem;
            width: 1rem;
        }

        --text: ${colors.primary.default};
        --text-active: ${colors.black.default};
        --bg: transparent;
        --bg-active: transparent;
        --bg-disabled: transparent;

        &.-default,
        &.-primary,
        &.-danger,
        &.-white {
            color: var(--text);
            background-color: var(--bg);
            border-color: var(--bg);

            &.-outline {
                border-color: var(--outline, var(--bg));
                color: var(--outline-text, var(--bg));
            }

            &.-text {
                color: var(--text-text, var(--bg));
            }

            &:hover,
            &.active {
                background-color: var(--bg-active);
                border-color: var(--bg-active);
                color: var(--text-active);

                &.-outline {
                    background-color: var(--bg) !important;
                    border: 1px solid var(--bg);
                    color: var(--outline-text-active, var(--text));
                }

                &.-text {
                    color: var(--text-text-active, var(--bg-active));
                }
            }

            &:disabled,
            &.disabled {
                background-color: var(--bg-disabled);
                border-color: var(--bg-disabled);
                color: var(--text);

                &.-outline {
                    border: 1px solid var(--outline-disabled, var(--outline-text-disabled, var(--bg-disabled)));
                    color: var(--outline-text-disabled, var(--bg-disabled));
                }

                &.-text {
                    color: var(--text-text-disabled, var(--bg-disabled));
                }
            }
        }

        &.-primary {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.primary.default};
            --bg-active: ${desaturate(darken(colors.primary.default))};
            --bg-disabled: ${desaturate(lighten(colors.primary.default))};
        }

        &.-danger {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.danger.default};
            --bg-active: ${desaturate(darken(colors.danger.default))};
            --bg-disabled: ${lighten(colors.danger.default, 0.3)};
        }

        &.-default {
            --text: ${colors.black.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.black.default};
            --outline-text: ${colors.black.default};
            --outline-text-active: ${colors.black.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.black.default};
            --text-text-active: ${colors.black.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-white {
            --text: ${colors.primary.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.white.default};
            --outline-text: ${colors.white.default};
            --outline-text-active: ${colors.primary.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.white.default};
            --text-text-active: ${colors.white.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-text {
            background-color: transparent !important;
            border: 1px solid transparent;
            box-shadow: none;

            &:hover,
            &.active {
                border-color: transparent;
            }

            &.disabled,
            &:disabled {
                border: 1px solid transparent;
            }
        }

        &.-outline {
            background-color: transparent !important;
            border: 1px solid inherit;
            box-shadow: none;
        }

        &.disabled,
        &:disabled {
            cursor: not-allowed;
        }
    `,
	({ size }) => {
		switch (size) {
		case 'small':
			return css`
                font-size: 0.9rem;
                padding: .2rem 1rem;
            `;
		case 'large':
			return css`
                font-size: 1.2rem;
                padding: 0.8rem 1.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	},

	({ block }) => block && css`
        display: block;
        width: 100%;
    `,

	({ loading }) => loading && css`
        color: transparent !important;
        cursor: wait !important;

        &:hover {
            color: transparent !important;
        }
    `,
);

const IconWrapper = styled.span`
    font-size: inherit;
    margin-right: 0.3rem;
    vertical-align: middle;

    svg {
        height: 1rem;
        width: 1rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 0.9rem;
                width: 0.9rem;
            `;
		case 'large':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}
    }
`;

const LoadingWrapper = styled.div`
    position: absolute;
    left: 50%;
    transform: translate(-50%, -50%);
    top: 50%;
    align-content: center;
    justify-content: center;
    display: flex;

    svg {
        height: 1.4rem;
        width: 1.4rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'large':
			return css`
                height: 2.4rem;
                width: 2.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}

    path {
        ${({ type, outline, text }) => {
		switch (type) {
		case 'primary':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.primary.default};
                `}

                ${text && css`
                    fill: ${colors.primary.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.primary.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}
                }
            `;
		case 'danger':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.danger.default};
                `}

                ${text && css`
                    fill: ${colors.danger.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.danger.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}
                }
            `;
		case 'default':
		default:
			return css`
                fill: ${colors.black.default};

                ${outline && css`
                    fill: ${colors.black.default};
                `}

                ${text && css`
                    fill: ${colors.black.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.black.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.black.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    fill: ${darken(colors.gray.default, 0.1)};

                    ${outline && css`
                        fill: ${lighten(colors.gray.default, 0.01)};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.black.default))};
                    `}
                }
            `;
		}}
}
        }
    }
`;

const renderIcon = (Icon) => {
	if (typeof Icon === 'function') {
		return <Icon />;
	}

	return Icon;
};

const ButtonWrapper = ({
	className,
	type,
	outline,
	text,
	size,
	icon,
	children, 
	component,
	loading,
	buttonType = 'button',
	...rest
}) => {
	let Component = Button;

	if (component) {
		Component = Button.withComponent(component);
	}

	return (
		<Component
			className={cx(className, {
				[`-${type}`]: type,
				'-outline': outline,
				'-text': text,
			})}
			outline={outline}
			text={text}
			size={size}
			loading={loading}
			type={buttonType}
			{...rest}
		>
			{
				icon && (
					<IconWrapper size={size}>{renderIcon(icon)}</IconWrapper>
				)
			}
			{
				loading && (
					<LoadingWrapper
						type={type}
						outline={outline}
						text={text}
						size={size}
						{...rest}
					>
						<Loading />
					</LoadingWrapper>
				)
			}
			{children}
		</Component>
	);
};

ButtonWrapper.propTypes = {
	block: bool,
	buttonType: string,
	component: oneOfType([string, func]),
	icon: oneOfType([node, func]),
	loading: bool,
	outline: bool,
	size: oneOf(['small', 'default', 'large']),
	text: bool,
	type: oneOf(['default', 'white', 'primary', 'danger', 'success']),
};

ButtonWrapper.defaultProps = {
	type: 'default',
};

export { Button as EmotionButton };
export default ButtonWrapper;
