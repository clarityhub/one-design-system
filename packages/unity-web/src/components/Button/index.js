export { default, EmotionButton } from './Button';