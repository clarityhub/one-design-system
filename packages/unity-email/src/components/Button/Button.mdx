---
menu: Components
name: Button
---

import { Playground, PropsTable } from 'docz';
import Button from './Button';

# Email Button

```js
import Button from '@clarityhub/unity-email/lib/components/Button';
```

## Prop Types

<PropsTable of={Button} />
