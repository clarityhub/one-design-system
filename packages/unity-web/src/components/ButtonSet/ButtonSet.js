import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool } from 'prop-types';
import { EmotionButton } from '../Button';

const ButtonSet = styled.div`
    ${({ spread }) => spread && css`
        display: flex;
        align-items: center;
        justify-content: flex-end;

        ${EmotionButton} {
            &:not(:first-of-type) {
                margin-left: auto!important;
            }
        }
    `}

    button + button,
    a + a,
    a + button,
    button + a {
        margin-left: 0.4rem;
    }
`;

/**
 * Hack for docz and react-docgen
 */
const ButtonSetPropTypes = () => <div />;
ButtonSetPropTypes.propTypes = {
	spread: bool,
};
ButtonSet.propTypes = ButtonSetPropTypes.propTypes;
export { ButtonSetPropTypes };

export default ButtonSet;

