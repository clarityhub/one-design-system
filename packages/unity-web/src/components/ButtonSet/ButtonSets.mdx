---
menu: Components
name: ButtonSets
---

import { Playground, PropsTable } from 'docz';
import Button from '../Button';
import ButtonSet, { ButtonSetPropTypes } from './ButtonSet';
import Typography from '../Typography';

# ButtonSets

A `ButtonSet` is used to group a set of action buttons together
and provide the appropriate spacing between them to make each button distinct, but
still visually grouped together.

```js
import ButtonSet from '@clarityhub/unity-web/lib/components/ButtonSet';
```

## Wrapping Buttons

<Playground>
    <Typography>Use Profile</Typography>
    <ButtonSet>
        <Button text>Back</Button>
        <Button type="primary">Edit</Button>
        <Button outline type="danger">Delete</Button>
    </ButtonSet>
    <br />
    <br />
    <Typography>Let's create a new user!</Typography>
    <ButtonSet>
        <Button>Cancel</Button>
        <Button type="primary">Create</Button>
    </ButtonSet>
    <br />
    <br />
    <Typography>Are you sure you want to delete this user?</Typography>
    <ButtonSet>
        <Button>Cancel</Button>
        <Button type="danger">Delete</Button>
    </ButtonSet>
</Playground>

## Spreading Buttons

<Playground>
    <ButtonSet spread>
        <Button text>Back</Button>
        <Button type="primary">Edit</Button>
        <Button outline type="danger">Delete</Button>
    </ButtonSet>
    <br />
    <br />
    <ButtonSet spread>
        <Button>Cancel</Button>
        <Button type="primary">Create</Button>
    </ButtonSet>
    <br />
    <br />
    <ButtonSet spread>
        <Button>Cancel</Button>
        <Button type="danger">Delete</Button>
    </ButtonSet>
</Playground>

## Prop Types

<PropsTable of={ButtonSetPropTypes} />
