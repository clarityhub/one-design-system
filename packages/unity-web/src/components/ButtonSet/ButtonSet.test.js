import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';
import ButtonSet from './ButtonSet';

describe('<ButtonSet />', () => {
	it('renders a set of buttons', () => {
		const node = (
			<ButtonSet>
				<Button text>Back</Button>
				<Button type="primary">Edit</Button>
				<Button outline type="danger">Delete</Button>
			</ButtonSet>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});


	it('renders a spread of buttons', () => {
		const node = (
			<ButtonSet spread>
				<Button text>Back</Button>
				<Button type="primary">Edit</Button>
				<Button outline type="danger">Delete</Button>
			</ButtonSet>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});