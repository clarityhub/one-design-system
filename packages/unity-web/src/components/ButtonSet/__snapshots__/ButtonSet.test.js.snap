// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSet /> renders a set of buttons 1`] = `
.emotion-6 button + button,
.emotion-6 a + a,
.emotion-6 a + button,
.emotion-6 button + a {
  margin-left: 0.4rem;
}

.emotion-0 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus,
.emotion-0.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-0:disabled,
.emotion-0.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-0 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-0.-default,
.emotion-0.-primary,
.emotion-0.-danger,
.emotion-0.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-0.-default.-outline,
.emotion-0.-primary.-outline,
.emotion-0.-danger.-outline,
.emotion-0.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-0.-default.-text,
.emotion-0.-primary.-text,
.emotion-0.-danger.-text,
.emotion-0.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-0.-default:hover,
.emotion-0.-primary:hover,
.emotion-0.-danger:hover,
.emotion-0.-white:hover,
.emotion-0.-default.active,
.emotion-0.-primary.active,
.emotion-0.-danger.active,
.emotion-0.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-0.-default:hover.-outline,
.emotion-0.-primary:hover.-outline,
.emotion-0.-danger:hover.-outline,
.emotion-0.-white:hover.-outline,
.emotion-0.-default.active.-outline,
.emotion-0.-primary.active.-outline,
.emotion-0.-danger.active.-outline,
.emotion-0.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-0.-default:hover.-text,
.emotion-0.-primary:hover.-text,
.emotion-0.-danger:hover.-text,
.emotion-0.-white:hover.-text,
.emotion-0.-default.active.-text,
.emotion-0.-primary.active.-text,
.emotion-0.-danger.active.-text,
.emotion-0.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-0.-default:disabled,
.emotion-0.-primary:disabled,
.emotion-0.-danger:disabled,
.emotion-0.-white:disabled,
.emotion-0.-default.disabled,
.emotion-0.-primary.disabled,
.emotion-0.-danger.disabled,
.emotion-0.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-0.-default:disabled.-outline,
.emotion-0.-primary:disabled.-outline,
.emotion-0.-danger:disabled.-outline,
.emotion-0.-white:disabled.-outline,
.emotion-0.-default.disabled.-outline,
.emotion-0.-primary.disabled.-outline,
.emotion-0.-danger.disabled.-outline,
.emotion-0.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-0.-default:disabled.-text,
.emotion-0.-primary:disabled.-text,
.emotion-0.-danger:disabled.-text,
.emotion-0.-white:disabled.-text,
.emotion-0.-default.disabled.-text,
.emotion-0.-primary.disabled.-text,
.emotion-0.-danger.disabled.-text,
.emotion-0.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-0.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-0.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-0.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #566ad2;
  --text-text-active: rgb(70,90,197);
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-0.-text:hover,
.emotion-0.-text.active {
  border-color: transparent;
}

.emotion-0.-text.disabled,
.emotion-0.-text:disabled {
  border: 1px solid transparent;
}

.emotion-0.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-0.disabled,
.emotion-0:disabled {
  cursor: not-allowed;
}

<div
  className="emotion-6 emotion-7"
>
  <button
    className="-default -text emotion-0 emotion-1"
    type="button"
  >
    Back
  </button>
  <button
    className="-primary emotion-0 emotion-1"
    type="button"
  >
    Edit
  </button>
  <button
    className="-danger -outline emotion-0 emotion-1"
    type="button"
  >
    Delete
  </button>
</div>
`;

exports[`<ButtonSet /> renders a spread of buttons 1`] = `
.emotion-0 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus,
.emotion-0.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-0:disabled,
.emotion-0.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-0 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-0.-default,
.emotion-0.-primary,
.emotion-0.-danger,
.emotion-0.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-0.-default.-outline,
.emotion-0.-primary.-outline,
.emotion-0.-danger.-outline,
.emotion-0.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-0.-default.-text,
.emotion-0.-primary.-text,
.emotion-0.-danger.-text,
.emotion-0.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-0.-default:hover,
.emotion-0.-primary:hover,
.emotion-0.-danger:hover,
.emotion-0.-white:hover,
.emotion-0.-default.active,
.emotion-0.-primary.active,
.emotion-0.-danger.active,
.emotion-0.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-0.-default:hover.-outline,
.emotion-0.-primary:hover.-outline,
.emotion-0.-danger:hover.-outline,
.emotion-0.-white:hover.-outline,
.emotion-0.-default.active.-outline,
.emotion-0.-primary.active.-outline,
.emotion-0.-danger.active.-outline,
.emotion-0.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-0.-default:hover.-text,
.emotion-0.-primary:hover.-text,
.emotion-0.-danger:hover.-text,
.emotion-0.-white:hover.-text,
.emotion-0.-default.active.-text,
.emotion-0.-primary.active.-text,
.emotion-0.-danger.active.-text,
.emotion-0.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-0.-default:disabled,
.emotion-0.-primary:disabled,
.emotion-0.-danger:disabled,
.emotion-0.-white:disabled,
.emotion-0.-default.disabled,
.emotion-0.-primary.disabled,
.emotion-0.-danger.disabled,
.emotion-0.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-0.-default:disabled.-outline,
.emotion-0.-primary:disabled.-outline,
.emotion-0.-danger:disabled.-outline,
.emotion-0.-white:disabled.-outline,
.emotion-0.-default.disabled.-outline,
.emotion-0.-primary.disabled.-outline,
.emotion-0.-danger.disabled.-outline,
.emotion-0.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-0.-default:disabled.-text,
.emotion-0.-primary:disabled.-text,
.emotion-0.-danger:disabled.-text,
.emotion-0.-white:disabled.-text,
.emotion-0.-default.disabled.-text,
.emotion-0.-primary.disabled.-text,
.emotion-0.-danger.disabled.-text,
.emotion-0.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-0.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-0.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-0.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #566ad2;
  --text-text-active: rgb(70,90,197);
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-0.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-0.-text:hover,
.emotion-0.-text.active {
  border-color: transparent;
}

.emotion-0.-text.disabled,
.emotion-0.-text:disabled {
  border: 1px solid transparent;
}

.emotion-0.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-0.disabled,
.emotion-0:disabled {
  cursor: not-allowed;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-6 .emotion-1:not(:first-of-type) {
  margin-left: auto!important;
}

.emotion-6 button + button,
.emotion-6 a + a,
.emotion-6 a + button,
.emotion-6 button + a {
  margin-left: 0.4rem;
}

<div
  className="emotion-6 emotion-7"
>
  <button
    className="-default -text emotion-0 emotion-1"
    type="button"
  >
    Back
  </button>
  <button
    className="-primary emotion-0 emotion-1"
    type="button"
  >
    Edit
  </button>
  <button
    className="-danger -outline emotion-0 emotion-1"
    type="button"
  >
    Delete
  </button>
</div>
`;
